pyiron_base/generic/units.py

```diff
@@ -0,0 +1,300 @@
+        """
+        if item in self._unit_dict.keys():
+            return self._unit_dict[item]
+        elif item in self._quantity_dict.keys():
+            return self._unit_dict[self._quantity_dict[item]]
+        else:
+            raise ValueError("Quantity/label '{}' not registered in this unit registry".format(item))
+
+    def get_dtype(self, quantity):
+        """
+        Returns the data type in which the quantity will be stored
+
+        Args:
+            quantity (str): The quantity
+
+        Returns:
+            type: Corresponding data type
+        """
+        if quantity in self._unit_dict.keys():
+            return self._dtype_dict[quantity]
+        elif quantity in self._quantity_dict.keys():
+            return self._dtype_dict[self._quantity_dict[quantity]]
+        else:
+            raise ValueError("Quantity/label '{}' not registered in this unit registry".format(quantity))
+
+
+class UnitConverter:
+    """
+
+    Module to handle conversions between two different unit registries mainly use to convert units between codes and
+    pyiron submodules.
+
+    To instantiate this class, you need two units registries: a base units registry and a code registry:
+
+    >>> import pint
+    >>> pint_registry = pint.UnitRegistry()
+    >>> base_units = PyironUnitRegistry()
+    >>> base_units.add_quantity(quantity="energy", unit=pint_registry.eV)
+    >>> code_units = PyironUnitRegistry()
+    >>> code_units.add_quantity(quantity="energy",
+    >>>                         unit=1e3 * pint_registry.cal / (pint_registry.mol * pint_registry.N_A))
+    >>> unit_converter = UnitConverter(base_units=base_units, code_units=code_units)
+
+    The unit converter instance can then be used to obtain conversion factors between code and base units either as a
+    `pint` quantity:
+
+    >>> print(unit_converter.code_to_base("energy"))
+    0.043364104241800934 electron_volt
+
+    or as a scalar:
+
+    >>> print(unit_converter.code_to_base_value("energy"))
+    0.043364104241800934
+
+    Alternatively, the unit converter can also be used as decorators for functions that return an array scaled into
+    appropriate units:
+
+    >>> @unit_converter(quantity="energy", conversion="code_to_base")
+    >>>    def return_ones():
+    >>>        return np.ones(5)
+    >>> print(return_ones())
+    [0.0433641 0.0433641 0.0433641 0.0433641 0.0433641]
+
+    The decorator can also be used to assign units for numpy arrays
+    (for more info see https://pint.readthedocs.io/en/0.10.1/numpy.html)
+
+    >>> @unit_converter(quantity="energy", conversion="base_units")
+    >>> def return_ones_ev():
+    >>>     return np.ones(5)
+    >>> print(return_ones_ev())
+    [1.0 1.0 1.0 1.0 1.0] electron_volt
+
+    """
+
+    def __init__(self, base_units, code_units):
+        """
+
+        Args:
+            base_units (pyiron_base.generic.units.PyironUnitRegistry): Base unit registry
```

Comment:
```suggestion
            base_units (:class:`pyiron_base.generic.units.PyironUnitRegistry`): Base unit registry
```
I think this is the right syntax to make links in the sphinx output.

Reply:
At least for references to our own classes we should use it, to make the docs more readable.